### PR TITLE
cleanup dead code of generating binding permanent id

### PR DESCRIPTION
**Pull request by whitewindmills**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Since the PR(https://github.com/karmada-io/karmada/pull/4474) has been merged, we're going to remove those codes which generate binding permanent ID outside of webhook.
Same as https://github.com/karmada-io/karmada/pull/4928

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @whitewindmills the E2E is failed.

**Comment by whitewindmills** [comment]
seems to be an unrelated e2e case.
/retest

**Comment by XiShanYongYe-Chang** [comment]
Hi @whitewindmills I want to confirm that the logic for generating permanent-id in the controller is removed in this version, which does not affect the upgrade from version 1.8 to version 1.9?

**Comment by whitewindmills** [comment]
> which does not affect the upgrade from version 1.8 to version 1.9?

I don't think it will. this commit will only be released after v1.10+.

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/lgtm

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/assign

File: pkg/dependenciesdistributor/dependencies_distributor.go [hunk 1]
```diff
@@ -653,8 +650,8 @@ func buildAttachedBinding(binding *workv1alpha2.ResourceBinding, object *unstruc
 		Clusters:  binding.Spec.Clusters,
 	})
 
-	policyID := util.GetLabelValue(binding.Labels, workv1alpha2.ResourceBindingPermanentIDLabel)
-	dependedLabels = util.DedupeAndMergeLabels(dependedLabels, map[string]string{bindingDependedIDLabelKey: policyID})
+	bindingID := util.GetLabelValue(binding.Labels, workv1alpha2.ResourceBindingPermanentIDLabel)
```
Comment (RainbowMango) [inline pkg/dependenciesdistributor/dependencies_distributor.go:653]: nice catch!

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**